File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int redLast = 0;
        int whiteLast = 0;
        int blueFirst = nums.size() - 1;

        while (whiteLast <= blueFirst) {
            if (nums[whiteLast] == 0) {
                swap(nums[whiteLast++], nums[redLast++]);
                continue;
            } 

            if (nums[whiteLast] == 1) {
                whiteLast++;
                continue;
            }

            swap(nums[whiteLast], nums[blueFirst--]);
        }
    }
};

File: 0075-sort-colors/0075-sort-colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        curr = 0
        blue = len(nums) - 1
        red = 0
        while curr <= blue:
            if nums[curr] == 0:
                tmp = nums[curr]
                nums[curr] = nums[red]
                nums[red] = tmp
                curr += 1
                red += 1
            elif nums[curr] == 1:
                curr += 1
            else:
                tmp = nums[curr]
                nums[curr] = nums[blue]
                nums[blue] = tmp
                blue -= 1
        